### ISSUE TITLE
minimum loss is always set to training loss

### ISSUE THREAD

**Issue by mheilman** [tensorflow/skflow#131]
The code in `BaseMonitor.update` currently sets the minimum loss so far as follows ([link](https://github.com/tensorflow/skflow/blob/master/skflow/monitors.py#L77)):

```
        if self.last_loss_seen < self.min_loss:
            self.min_loss = training_loss
            self.min_loss_i = self.steps
```

I believe the 2nd line there should be `self.min_loss = self.last_loss_seen` since the validation monitor sets that [here](https://github.com/tensorflow/skflow/blob/master/skflow/monitors.py#L158).

**Comment by terrytangyuan** [comment]
@mheilman Thanks for spotting that out! I just pushed the necessary change. You are welcome to submit PR directly to fix issues you found in the future once you signed CLA!

**Comment by mheilman** [comment]
You're welcome.  Thanks for the quick fix.  If I happen find anything else in the future, I'll submit a PR (once the CLA stuff is sorted out).
